ldk/go/docs/loops.md

```diff
@@ -1,51 +1,108 @@
-# Loops
+# An Example of a simple loop with Window State listener
 
-Loops receive events and use them to generate relevant whispers. Loops choose which events they want to use and which they want to ignore.
+```go
+package main
 
-## Interface
+import (
+	"context"
 
-Writing a loop boils down to writing an implementation for the `Loop` interface.
+	ldk "github.com/open-olive/loop-development-kit/ldk/go"
+)
 
-```go
-type Loop interface {
-	LoopStart(Sidekick) error
-	LoopStop() error
+// main function to serve the loop
+func main() {
+	logger := ldk.NewLogger("demo-loop-go")
+	var loop ldk.Loop
+	loop = &Loop{
+		logger:         logger,
+	}
+
+	// serve the loop implementation
+	ldk.ServeLoopPlugin(logger, loop)
 }
-```
 
-**LoopStart** - The Loop should wait to start operating until this is called. The provided `Sidekick` reference should be stored in memory for continued use.
+// define structure of loop
+type Loop struct {
+	ctx              context.Context
+	cancel           context.CancelFunc
+	sidekick         ldk.Sidekick
+	logger           *ldk.Logger
+}
 
-**LoopStop** - The Loop should stop operating when this is called.
+// LoopStart will be called by ldk when loop starts
+func (loop *Loop) LoopStart(sidekick ldk.Sidekick) error {
```

Comment:
Go's [code style](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names) prefers that receiver variable names are one or two character

Reply:
Understood.